pkg/cloudprovider/cloudapi/azure/azure_compute.go

```diff
@@ -261,22 +263,25 @@ func (computeCfg *computeServiceConfig) DoResourceInventory() error {
 	return nil
 }
 
-func (computeCfg *computeServiceConfig) SetResourceFilters(selector *crdv1alpha1.CloudEntitySelector) {
+func (computeCfg *computeServiceConfig) AddSelectors(selector *crdv1alpha1.CloudEntitySelector) error {
 	subscriptionIDs := []string{computeCfg.credentials.SubscriptionID}
 	tenantIDs := []string{computeCfg.credentials.TenantID}
 	locations := []string{computeCfg.credentials.region}
+
 	if filters, found := convertSelectorToComputeQuery(selector, subscriptionIDs, tenantIDs, locations); found {
-		computeCfg.computeFilters[selector.GetName()] = filters
+		key := selector.GetNamespace() + "/" + selector.GetName()
+		computeCfg.computeFilters[key] = filters
+		computeCfg.selectors[key] = selector.DeepCopy()
 	} else {
-		if selector != nil {
-			delete(computeCfg.computeFilters, selector.GetName())
-		}
-		computeCfg.resourcesCache.UpdateSnapshot(nil)
+		// TODO: Set CES status with the following error.
```

Comment:
is it still TODO, if so, can you add a fix for this in this PR

Reply:
I did try implementing it and decided against taking it up right now, I see following reasons, we can discuss on.
1. We can add status field in CES CRD, set selector status and store in service config. However, the logic where account poller fetches the selector(selectors) status and set status of CR in etcd , is dependent on how we are going to implement multiple CES logic in account poller. 
2. Also, right now we set vpc and vm inventory poll status error is set as CPA error, this needs to be segregated and vm inventory poll error should be set in "Status" field of respective CES.

Because, I saw a strong dependency wrt the design we are going to have with multiple CES, I decided to take it up along with it.